### PR TITLE
Add periodic e2e tests for Cluster API AWS Provider

### PR DESCRIPTION
**Pull request by harishspqr**
The periodic tests run every hour. 

Related to: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/272 
I have taken over this PR from akutz: https://github.com/kubernetes/test-infra/pull/11524 

Blocked by:
    1. boskos: Deploying a new Boskos janitor for AWS (#11511): Waiting for the deployment of aws-janator-boskos in prow.

/hold

### CODE REVIEW

**Comment by ncdc** [comment]
@harishspqr thanks for doing this! I know you & I had talked with Andrew about removing the postsubmit, but after some discussions on Slack, let's keep the postsubmit. It will provide us with useful information about AWS account usage so we can better understand if it would be possible to run e2e smoke tests in presubmit jobs.

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 periodics:  
-- interval: 4h
-  name: periodic-cluster-api-provider-aws-test-creds
+- name: periodic-cluster-api-provider-aws-test-creds
```
Comment (harishspqr) [inline config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: I am not entirely sure if the periodic AWS test creds must run. It verifies if the preset creds can create/delete resources.

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml [hunk 2]
```diff
@@ -15,4 +17,57 @@ periodics:
...
+      - "./scripts/ci-aws-cred-test.sh"
+      securityContext:
+        privileged: true
+- name: periodic-cluster-api-provider-aws-bazel-e2e
+  decorate: true
+  interval: 1h
+  labels:
+    preset-dind-enabled: "true"
+    preset-kind-volume-mounts: "true"
+    preset-service-account: "true"
```
Comment (harishspqr) [inline config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: I am un-sure if we need the preset aws-ssh and preset aws-credentials for this job. Please verify.

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml (hunk 1 above)
Comment (ncdc) [reply config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: Let's not remove anything that's already there for now

**Comment by harishspqr** [comment]
@k8s-ci-robot I signed CLA

**Comment by harishspqr** [comment]
@k8s-ci-robot I signed CLA

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml [hunk 3]
```diff
@@ -15,4 +15,55 @@ periodics:
...
+      - e2e
+      env:
+      - name: BOSKOS_HOST
+        value: "http://boskos.test-pods.svc.cluster.local."
+      securityContext:
+        privileged: true
+postsubmits:
+  kubernetes-sigs/cluster-api-provider-aws:
+  - name: pull-cluster-api-provider-aws-bazel-e2e
+    decorate: true
```
Comment (ncdc) [inline config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: I think you also need `path_alias: "sigs.k8s.io/cluster-api-provider-aws"` here (compare to the jobs in https://github.com/kubernetes/test-infra/blob/17cf3f083a84bbd4cb48212f5985f55cc36141a7/config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-presubmits.yaml).

File: testgrid/config.yaml [hunk 4]
```diff
@@ -6092,6 +6098,10 @@ dashboards:
     test_group_name: pull-cluster-api-provider-aws-bazel-integration
   - name: test-creds
     test_group_name: periodic-cluster-api-provider-aws-test-creds
+  - name: periodic-bazel-e2e
+    test_group_name: periodic-cluster-api-provider-aws-bazel-e2e
+  - name: pr-bazel-e2e
```
Comment (ncdc) [inline testgrid/config.yaml]: To match the other names (bazel-pr-verify, bazel-pr-build, etc), could we call this bazel-pr-e2e?

File: testgrid/config.yaml [hunk 5]
```diff
@@ -6092,6 +6098,10 @@ dashboards:
     test_group_name: pull-cluster-api-provider-aws-bazel-integration
   - name: test-creds
     test_group_name: periodic-cluster-api-provider-aws-test-creds
+  - name: periodic-bazel-e2e
```
Comment (ncdc) [inline testgrid/config.yaml]: Should we call this bazel-periodic-e2e?

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml [hunk 6]
```diff
@@ -10,9 +10,61 @@ periodics:
   - org: kubernetes-sigs
     repo: cluster-api-provider-aws
     base_ref: master
-    path_alias: "sigs.k8s.io/cluster-api-provider-aws"
+    path_alias: sigs.k8s.io/cluster-api-provider-aws
```
Comment (ncdc) [inline config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: Why this change?

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml [hunk 7]
```diff
@@ -10,9 +10,61 @@ periodics:
...
+- name: periodic-cluster-api-provider-aws-bazel-e2e
+  decorate: true
+  interval: 1h
+  labels:
+    preset-dind-enabled: "true"
+    preset-kind-volume-mounts: "true"
+    preset-service-account: "true"
+    preset-aws-ssh: "true"
+    preset-aws-credential: "true"
+  extxa_refs:
```
Comment (detiber) [inline config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: Should this be `extra_refs` instead?

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml [hunk 8]
```diff
@@ -10,9 +10,61 @@ periodics:
...
+      args:
+      - e2e
+      env:
+      - name: BOSKOS_HOST
+        value: "http://boskos.test-pods.svc.cluster.local."
+      securityContext:
+        privileged: true
+postsubmits:
+  kubernetes-sigs/cluster-api-provider-aws:
+  - name: pull-cluster-api-provider-aws-bazel-e2e
```
Comment (detiber) [inline config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: This should probably be named something other than pull. maybe `ci-cluster-api-provider-aws-bazel-e2e`?

File: config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml (hunk 6 above)
Comment (harishspqr) [reply config/jobs/kubernetes-sigs/cluster-api-provider-aws/cluster-api-provider-aws-ci.yaml]: not intentional but seems like it doesn't matter.

**Comment by detiber** [comment]
Removing the hold since the aws-janitor has now been deployed
/hold cancel

**Comment by ncdc** [comment]
I see Aaron is out... would anyone else be willing to approve? @michelle192837 @wojtek-t @shyamjvs

**Comment by detiber** [comment]
@harishspqr It looks like the job is now running :tada:, however it is currently failing on trying to retrieve the account from Boskos: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/periodic-cluster-api-provider-aws-bazel-e2e/1113430504989790208

**Comment by ncdc** [comment]
We need to change the `BOSKOS_HOST` values from `value: "http://boskos.test-pods.svc.cluster.local."` to `value: "boskos.test-pods.svc.cluster.local"`
